### ISSUE TITLE
Automatic Status Updates Not Working

### ISSUE THREAD

**Issue by chobo997** [IndigoDomotics/indigo-ecobee-plugin#11]
Have tried several versions of the new and improved Ecobee plugin (now using v1.2.8), and for some reason the thermostat does not update its status unless I click "Send Status Request" in the Indigo client GUI.

Automatic status updates were working with earlier "jdhorne" versions of the plugin. I have deleted my original Ecobee Thermostat device and re-added it thinking that might resolve it — no luck.

Control works fine along with manual status requests. Modifying the update frequency in the plugin config doesn't change things for the better. (BTW: is this minutes or seconds — text isn't terribly clear on this?)

Indigo 7.1.1 and macOS 10.9.5 on a Mac mini

**Comment by FlyingDiver** [comment]
The temps don't update at all?  Or just not the "status" of the device?

**Comment by chobo997** [comment]
Nothing updates automatically (temps/modes/status, etc.). This is now in the log:

> Ecobee                          Update failed, Ecobee servers returned error code 500
>    Ecobee Error                    Error in plugin execution GetUiDynamicList:
> 
> Traceback (most recent call last):
>   File "plugin.py", line 240, in get_thermostats
>   File "/Library/Application Support/Perceptive Automation/Indigo 7/Plugins/Ecobee.indigoPlugin/Contents/Server Plugin/ecobee_devices.py", line 48, in get_thermostats
> TypeError: 'NoneType' object is not iterable

**Comment by FlyingDiver** [comment]
What specific Ecobee devices do you have?

**Comment by chobo997** [comment]
Original Ecobee3 and three remote sensors.

**Comment by FlyingDiver** [comment]
OK, thanks.  I thought I had killed that bug in 1.2.8, but apparently not.

**Comment by chobo997** [comment]
Got it. Will stay tuned for next steps.

**Comment by chobo997** [comment]
Quick note: had some time to remove and add back the thermostat and remote sensors this evening. Once they were added back I requested a new pin, added the app via the website, and restarted the plugin. It's now working (fingers crossed).

One other observation: changes to the update frequency don't take effect until after a plugin restart. Perhaps the plugin should restart automatically after clicking "Save" in the configure dialog box?
